moveit_planners/trajopt/include/kinematic_terms.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+#include <trajopt_utils/macros.h>
+TRAJOPT_IGNORE_WARNINGS_PUSH
+#include <Eigen/Core>
+TRAJOPT_IGNORE_WARNINGS_POP
+
+#include <trajopt/common.hpp>
+#include <trajopt_sco/modeling.hpp>
+#include <trajopt_sco/modeling_utils.hpp>
+#include <trajopt_sco/sco_fwd.hpp>
+#include <trajopt_sco/num_diff.hpp>
+
+#include <moveit/planning_scene/planning_scene.h>
+
+#include <Eigen/Geometry>
+
+using namespace Eigen;
+using namespace std;
+
+namespace trajopt_interface
+{
+
+Vector3d rotVec(const Matrix3d& m)
+{
+  Quaterniond q; q = m;
+  return Vector3d(q.x(), q.y(), q.z());
+}
+
+
+VectorXd concat(const VectorXd& a, const VectorXd& b)
+{
+  VectorXd out(a.size()+b.size());
+  out.topRows(a.size()) = a;
+  out.middleRows(a.size(), b.size()) = b;
+  return out;
+}
+
+template <typename T>
+vector<T> concat(const vector<T>& a, const vector<T>& b)
+{
+  vector<T> out;
+  vector<int> x(a.size() + b.size());
+  out.insert(out.end(), a.begin(), a.end());
+  out.insert(out.end(), b.begin(), b.end());
+  return out;
+}
+
+
+/**
+ * @brief Used to calculate the error for StaticCartPoseTermInfo
+ * This is converted to a cost or constraint using TrajOptCostFromErrFunc or TrajOptConstraintFromErrFunc
+ */
+struct CartPoseErrCalculator : public sco::VectorOfVector
```

Comment:
Have you used rviz_visual_tools to better understand if your vectors are in the right frame, and if the error calculation intuitively makes sense?

Reply:
I've never seen a struct inheritance... this isn't normal

Reply:
I got it from the original trajopt pakage. Should I change them to class?

Reply:
I am actually using rviz_visual_tools

Reply:
yes, per Google style, if it has functions, it should be a class